File: Python Exams/Basic Exam Collection/Basics Online Exam - 6 and 7 July 2019/6. TheMostPowerfulWord/6. TheMostPowerfulWord.py
from math import floor
current_word = input()
collection_list = ['a', 'e', 'i', 'o', 'u', 'y' ,'A', 'E', 'I', 'O','U', 'Y']
most_power_word = ''
most_power_sum = 0

while current_word != 'End of words':

    first_letter = current_word[0:1]
    other_letters_sum = sum([ord(x) for x in current_word])

    if first_letter in collection_list:
        other_letters_sum *= len(current_word)
    else:
        other_letters_sum =  floor(other_letters_sum / len(current_word))

    if most_power_sum <= other_letters_sum:
      most_power_sum = other_letters_sum
      most_power_word = current_word

    current_word = input()


print(f'The most powerful word is {most_power_word} - {most_power_sum}')



'''
But
Some
People
Say
It's
LOVE
End of words
'''

File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 6 and 7 July 2019/6. TheMostPowerfulWord/6. TheMostPowerfulWord.js
function theMostPowerfulWord(arg) {
  let sumWordChar = ''
  let totalSum = 0
  let powerWord = {}

  for (let i = 0; i < arg.length; i++) {
    if (arg[i] === 'End of words') {
      break
    }
    word = arg[i]
    sumWordChar = word.split('').map(el => el.charCodeAt(0)).reduce((a, b) => +a + +b)
    wordFirstChar = word.slice(0, 1)

    if (/[aeiouyAEIOUY]/gm.test(wordFirstChar)) {
      totalSum = sumWordChar * word.length
    } else {
      totalSum = sumWordChar / word.length
    }

    powerWord[word] = Math.round(totalSum)
    totalSum = 0
  }

 const mostStrongWord =  Object.keys(powerWord).reduce((a , b) => powerWord[a] > powerWord[b] ? a : b )
 console.log(`The most powerful word is ${mostStrongWord} - ${powerWord[mostStrongWord]}`);
}

//theMostPowerfulWord(['The', 'Most', 'Powerful', 'Word', 'Is', 'Experience', 'End of words', ])
//theMostPowerfulWord(["But", "Some", "People", "Say", "It's", "LOVE", "End of words", ])
